sample-app/src/utils/searchhandler.ts

```diff
@@ -0,0 +1,68 @@
+import { AnswersActions } from "@yext/answers-headless-react";
+import { SearchIntent } from "@yext/answers-headless";
+
+
+const defaultGeolocationOptions: PositionOptions = {
+  enableHighAccuracy: false,
+  timeout: 6000,
+  maximumAge: 300000,
+};
+
+/**
+ * Provide utility functions related to vertical and universal search execution.
+ * This includes any potential data check or setup related to a query before
+ * performing a search, such as fetching searchIntents using autocomplete requests,
+ * and retrieving user's location using nagivator.geolocation API.
+ */
+export default class SearchHandler {
+  static async executeSearch(answersActions: AnswersActions, isVertical: boolean) {
+    isVertical
+      ? answersActions.executeVerticalQuery()
+      : answersActions.executeUniversalQuery();
+    return;
+  }
+
+  static async executeSearchWithIntents(
```

Comment:
I think it might be a little clearer to have this method separate from the other 3. In my mind, since it uses the other 3, `executeSearchWithIntents` is at a higher level of abstraction than the others. The other 3 are bricks while this one is... whatever you feel like making with bricks

Reply:
Like in a different file? Although it's at a higher level of abstraction, it's still a utility function for search execution so I think it make sense to group them? I imagined it's similar to appliedfilterutils, where there are separate `brick` functions but then there are ones like `getGroupedAppliedFilters` that uses other `brick` functions but overall is still consider a utility function for applied filters.

Reply:
ok sgtm! I liked the organization a lot before but this makes sense too